src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -68,6 +68,10 @@ public function process(ContainerBuilder $container)
                         '/[^a-z0-9]/i',
                     ), function ($matches) { return strtoupper($matches[0]); }, $event['event']);
                     $event['method'] = preg_replace('/[^a-z0-9]/i', '', $event['method']);
+
+                    if (null !== ($class = $container->getDefinition($id)->getClass()) && !method_exists($class, $event['method']) && method_exists($class, '__invoke')) {
```

Comment:
we should use $container->getReflectionClass() instead
this will fix the current logic that misses resource tracking